www/js/app.js

```diff
@@ -1300,8 +1317,75 @@ function displayFileSelect () {
         });
         globalDropZone.addEventListener('drop', handleFileDrop);
     }
-    // This handles use of the file picker
-    document.getElementById('archiveFiles').addEventListener('change', setLocalArchiveFromFileSelect);
+
+    if (!isFireFoxOsNativeFileApiAvailable) {
```

Comment:
This seems really counterintuitive, that you have to keep saying "only run this if a defunct OS is not supported! I realize it does the job, but wouldn't it be easier just to put the final legacy picker in its own function `useLegacyFilePicker()` and do something like:

```
if (isFireFoxOsNativeFileApiAvailable) {
    useLegacyFilePicker();
    return;
}

```
Then  you can get rid of this test and all the other FireFoxOS tests below. What do you think?